src/ast/passes/fold_literals.cpp

```diff
@@ -65,23 +65,42 @@ static Expression make_boolean(ASTContext &ast, T left, T right, Binop &op)
       value = left || right;
       break;
     case Operator::PLUS:
+      value = left + right;
+      break;
     case Operator::MINUS:
+      value = left - right;
+      break;
     case Operator::MUL:
+      value = left && right;
+      break;
     case Operator::DIV:
+      value = static_cast<bool>(left && right);
```

Comment:
This is just for literal folding, so the value will always be known. For both `DIV` and `MOD`, I think you need to surface as an error on the binop. (You can probably just leave the value here but add the error to the node passed in as a parameter if `!right`).

Reply:
Done.